modules/core/arrow-syntax/src/main/kotlin/arrow/syntax/collections/iterable.kt

```diff
@@ -1,18 +1,11 @@
 package arrow.syntax.collections
 
 import arrow.core.Option
-import arrow.core.Predicate
 import arrow.core.andThen
 import arrow.core.getOrElse
-import arrow.core.toOption
-
-@Deprecated(message = "`firstOption` is now part of the Foldable interface and generalized to all foldable data types")
-fun <T> Iterable<T>.firstOption(): Option<T> = firstOrNull().toOption()
-
-@Deprecated(message = "`firstOption` is now part of the Foldable interface and generalized to all foldable data types")
-fun <T> Iterable<T>.firstOption(predicate: Predicate<T>): Option<T> = firstOrNull(predicate).toOption()
 
 @Deprecated(message = "`collect` is now part of the FunctorFilter interface and generalized to all filterable data types")
+@Suppress("UNCHECKED_CAST")
```

Comment:
Is the cast necessay? What's the inferred type?

Reply:
```
f.andThen { optionB -> optionB.getOrElse { g(value) } } as (A) -> Option<B>
```
This Cast can't be type checked

Reply:
Why not? If you break it down and ask the IDE to fill in the types, what is the suggested result? There is nothing in the definition that cannot be checked.

Reply:
the type checker expects (A) -> Any?

Reply:
That's telling us that the implementation is incorrect. You can see that `optionB.getOrElse { }` returns `Any?` because if it's present then it returns `B`, whereas if it isn't it returns `Option<B>` and that's never flattened. `optionB.fold({ Some(it) }, { g(value) })`

Reply:
`f(value).map { listOf(it) }.getOrElse { emptyList() }` can also be simplified to a single `fold`